### PR TITLE
adding linking post

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7796
https://github.com/bcgov/entity/issues/8529
https://github.com/bcgov/entity/issues/8530

*Description of changes:*

1. POST endpoint for linking the routing slips
2. fixed some foreign keys

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Review by karthik-aot — APPROVED** [review]

File: pay-api/src/pay_api/services/fas/routing_slip.py [hunk 1]
```diff
@@ -323,6 +323,28 @@ def create(cls, request_json: Dict[str, any], **kwargs):
...
+
+        # do validations if its linkable
+        RoutingSlip._validate_linking(routing_slip=routing_slip, parent_rs_slip=parent_routing_slip)
+
+        routing_slip.parent_number = parent_routing_slip.number
+        routing_slip.status = RoutingSlipStatus.LINKED.value
+
+        # transfer the amount to parent.
+        # we keep the total amount as such and transfer only the remaining amount.
+        parent_routing_slip.remaining_amount += routing_slip.remaining_amount
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/fas/routing_slip.py:342]: dont we need to flush parent_routing_slip ?

File: pay-api/src/pay_api/services/fas/routing_slip.py [hunk 2]
```diff
@@ -338,3 +360,24 @@ def update(cls, rs_number: str, action: str, request_json: Dict[str, any]) -> Di
...
+        3). one of them has transactions
+        """
+        if routing_slip.parent:
+            raise BusinessException(Error.RS_ALREADY_HAS_A_PARENT)
+        children = RoutingSlipModel.find_children(routing_slip.number)
+        if len(children) > 0:
+            raise BusinessException(Error.RS_ALREADY_A_PARENT)
+
+        # has one of these has pending
+        if parent_rs_slip.invoices:
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/fas/routing_slip.py:379]: This I dont think is needed. A parent can have invoices afaik.

**Review by sumesh-aot — APPROVED** [review]
Please check comment

File: pay-api/migrations/versions/f841ac43e15e_new_rs_status_and_error_codes.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                'title': 'Routing Slip is already a parent.',
+                'detail': 'Linking cannot be performed since the routing is already a parent.'
+            },
+            {
+                'code': 'RS_ALREADY_HAS_A_PARENT',
+                'title': 'Routing Slip is already has a parent.',
+                'detail': 'Linking cannot be performed since the routing already has a parent.'
+            },
+            {
+                'code': 'RS_PARENT_HAS_TRANSACTIONS',
```
[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply pay-api/migrations/versions/f841ac43e15e_new_rs_status_and_error_codes.py:54]: > Please check comment

I wiill do another PR for fixing the PR comments

**Pull request merged**